e2e/internal/e2e/imageverify.go

```diff
@@ -0,0 +1,406 @@
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package e2e
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/sylabs/singularity/internal/pkg/test"
+)
+
+// ImageVerify checks for an image integrity
+func ImageVerify(t *testing.T, cmdPath string, imagePath string, labels bool, runDisabled bool) {
+	type testSpec struct {
+		name          string
+		execArgs      []string
+		expectSuccess bool
+	}
+	tests := []testSpec{
```

Comment:
Could these `/.singularity.d/` paths be put into `const` values somewhere? They seem to be repeating in a few places.

Perhaps a struct (`SingularityRoot`) could manage them?

```
sr = NewSingularityRoot('/.singularity.d')
test -f sr.GetRunScript()
test -f sr.GetEnvironmentFile()
```

What do you think?

Reply:
This could be a feature added after your PR.

Reply:
I think this would be best to add later on if we do extensive metadata related checks